runtime/gc_realtime/RealtimeMarkingScheme.cpp

```diff
@@ -480,502 +83,88 @@ void
-	fj9object_t *endScanPtr = startScanPtr + (((J9JavaVM *)env->getLanguageVM())->arrayletLeafSize / sizeof(fj9object_t));
-	return scanPointerRange(env, startScanPtr, endScanPtr);
 }
 
-MMINLINE UDATA
-MM_RealtimeMarkingScheme::scanPointerRange(MM_EnvironmentRealtime *env, fj9object_t *startScanPtr, fj9object_t *endScanPtr)
-{
-	fj9object_t *scanPtr = startScanPtr;
-	UDATA pointerFieldBytes = (UDATA)(endScanPtr - scanPtr);
-	UDATA pointerField = pointerFieldBytes / sizeof(fj9object_t);
-	while(scanPtr < endScanPtr) {
-		GC_SlotObject slotObject(_javaVM->omrVM, scanPtr);
-		markObject(env, slotObject.readReferenceFromSlot());
-		scanPtr++;
-	}
-
-	env->addScannedBytes(pointerFieldBytes);
-	env->addScannedPointerFields(pointerField);
-
-	return pointerField;
-}
-
-MMINLINE UDATA
-MM_RealtimeMarkingScheme::scanObject(MM_EnvironmentRealtime *env, J9Object *objectPtr)
-{
-	UDATA pointersScanned = 0;
-	switch(_gcExtensions->objectModel.getScanType(objectPtr)) {
-	case GC_ObjectModel::SCAN_ATOMIC_MARKABLE_REFERENCE_OBJECT:
-	case GC_ObjectModel::SCAN_MIXED_OBJECT:
-	case GC_ObjectModel::SCAN_OWNABLESYNCHRONIZER_OBJECT:
-	case GC_ObjectModel::SCAN_CLASS_OBJECT:
-	case GC_ObjectModel::SCAN_CLASSLOADER_OBJECT:
-		pointersScanned = scanMixedObject(env, objectPtr);
-		break;
-	case GC_ObjectModel::SCAN_POINTER_ARRAY_OBJECT:
-		pointersScanned = scanPointerArrayObject(env, (J9IndexableObject *)objectPtr);
-		break;
-	case GC_ObjectModel::SCAN_REFERENCE_MIXED_OBJECT:
-		pointersScanned = scanReferenceMixedObject(env, objectPtr);
-		break;
-	case GC_ObjectModel::SCAN_PRIMITIVE_ARRAY_OBJECT:
-	   pointersScanned = 0;
-	   break;
-	default:
-		Assert_MM_unreachable();
-	}
-	
-	return pointersScanned;
-}
-
-MMINLINE UDATA
-MM_RealtimeMarkingScheme::scanMixedObject(MM_EnvironmentRealtime *env, J9Object *objectPtr)
+void
+MM_RealtimeMarkingScheme::markLiveObjectsRoots(MM_EnvironmentBase *env)
 {
-	/* Object slots */
+	MM_EnvironmentRealtime *realtimeEnv = MM_EnvironmentRealtime::getEnvironment(env);
+	MM_MetronomeDelegate *metronomeDelegate = _realtimeGC->getRealtimeDelegate();
 
-	fj9object_t *scanPtr = _gcExtensions->mixedObjectModel.getHeadlessObject(objectPtr);
-	UDATA objectSize = _gcExtensions->mixedObjectModel.getSizeInBytesWithHeader(objectPtr);
-	fj9object_t *endScanPtr = (fj9object_t*)(((U_8 *)objectPtr) + objectSize);
-	UDATA *descriptionPtr;
-	UDATA descriptionBits;
-	UDATA descriptionIndex;
-#if defined(J9VM_GC_LEAF_BITS)
-	UDATA *leafPtr;
-	UDATA leafBits;
-#endif /* J9VM_GC_LEAF_BITS */
-	
-#if defined(J9VM_GC_DYNAMIC_CLASS_UNLOADING)
-	if(_realtimeGC->getRealtimeDelegate()->isDynamicClassUnloadingEnabled()) {
-		markClassOfObject(env, objectPtr);
-	}
-#endif /* J9VM_GC_DYNAMIC_CLASS_UNLOADING */
+	metronomeDelegate->markLiveObjectsRoots(realtimeEnv);
+	_scheduler->condYieldFromGC(realtimeEnv);
+	/* Heap Marking and barrier processing. Cannot delay barrier processing until the end.*/	
+	_realtimeGC->completeMarking(realtimeEnv);
```

Comment:
This is where markLiveObjectsScan should start (including this line).

Reply:
Again the model is base MarkingScheme (non-realtime). Over there markLiveObjectsScan, does completeMarking, what is the minimum core work. RT does a bit more.

Reply:
Okay, that's a simple fix. Done